ipld/unixfs/mod/dagmodifier_test.go

```diff
@@ -867,3 +872,655 @@ func getOffset(reader uio.DagReader) int64 {
+			Codec:    cid.DagProtobuf,
+			MhType:   mh.IDENTITY,
+			MhLength: -1,
+		})
+
+		// Verify initial data fits in identity
+		encodedInitial, _ := initialNode.EncodeProtobuf(false)
+		if len(encodedInitial) > verifcid.DefaultMaxIdentityDigestSize {
+			t.Skipf("Initial data too large for identity: %d bytes", len(encodedInitial))
+		}
+
+		err := dserv.Add(ctx, initialNode)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Create DagModifier
+		dmod, err := NewDagModifier(ctx, initialNode, dserv, testu.SizeSplitterGen(512))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Configure a custom prefix with specific values
+		dmod.Prefix = cid.Prefix{
+			Version:  1,
+			Codec:    cid.DagProtobuf,
+			MhType:   mh.SHA2_512,
+			MhLength: 32, // Truncated SHA-512
+		}
+
+		// Write within the existing data bounds to trigger modifyDag
+		// Use enough data that would cause identity overflow
+		modData := make([]byte, 50)
+		for i := range modData {
+			modData[i] = 'X'
+		}
+
+		// Write at offset 0, replacing first 50 bytes
+		n, err := dmod.WriteAt(modData, 0)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if n != len(modData) {
+			t.Fatalf("expected to write %d bytes, wrote %d", len(modData), n)
+		}
+
+		err = dmod.Sync()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Get the modified node
+		modifiedNode, err := dmod.GetNode()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// The node should still be using identity since the data is small
+		// But let's test that prefix fields would be preserved if it switched
+
+		// For this test, let's verify that our safePrefixForSize function
+		// correctly preserves fields when it would switch
+		testPrefix, changed := dmod.safePrefixForSize(initialNode.Cid().Prefix(), verifcid.DefaultMaxIdentityDigestSize+10)
+
+		if !changed {
+			t.Errorf("safePrefixForSize: expected prefix to change for oversized identity")
+		}
+		if testPrefix.Version != 1 {
+			t.Errorf("safePrefixForSize: expected Version 1, got %d", testPrefix.Version)
+		}
+		if testPrefix.Codec != cid.DagProtobuf {
+			t.Errorf("safePrefixForSize: expected Codec DagProtobuf, got %d", testPrefix.Codec)
+		}
+		if testPrefix.MhType != mh.SHA2_512 {
+			t.Errorf("safePrefixForSize: expected MhType SHA2_512 (%d), got %d", mh.SHA2_512, testPrefix.MhType)
+		}
+		if testPrefix.MhLength != 32 {
+			t.Errorf("safePrefixForSize: expected MhLength 32, got %d", testPrefix.MhLength)
+		}
+
+		// Also verify the actual node behavior (it should still be identity for this small data)
+		if modifiedNode.Cid().Prefix().MhType != mh.IDENTITY {
+			t.Logf("Note: Node switched from identity even though data is small")
+		}
+	})
+
+	t.Run("raw node preserves codec", func(t *testing.T) {
+		// Create a RawNode with identity CID that's near the limit
+		// This allows us to test overflow with small modifications
+		initialData := make([]byte, verifcid.DefaultMaxIdentityDigestSize-10)
+		for i := range initialData {
+			initialData[i] = byte('a' + i%26)
+		}
```

Comment:
See makeTestData above.
```suggestion
		initialData := makeTestData(verifcid.DefaultMaxIdentityDigestSize-10)
```